delphi/translators/for2py/pyTranslate.py

```diff
@@ -315,23 +323,44 @@ def proc_intrinsic(self, node):
 
         arg_list = self.get_arg_list(node)
         arg_strs = [
-            self.proc_expr(arg_list[i], False) for i in range(len(arg_list))
+            self.proc_expr(arg_list[i], False, printState) for i in range(len(arg_list))
         ]
 
         if py_mod != None:
             handler = f"{py_mod}.{py_fn}"
         else:
             handler = py_fn
 
+        for index, arg in enumerate(arg_list):
+            status = self.check_ref(index, arg, arg_strs)
+
         if py_fn_type == "FUNC":
             arguments = ", ".join(arg_strs)
-            return f"{handler}({arguments})"
+            return f"FloatNumpy({handler}({arguments}))"
         elif py_fn_type == "INFIXOP":
-            assert len(arg_list) == 2, f"INFIXOP with {len(arglist)} arguments"
+            assert len(arg_list) == 2, f"INFIXOP with {len(arg_list)} arguments"
             return f"({arg_strs[0]} {py_fn} {arg_strs[1]})"
         else:
             assert False, f"Unknown py_fn_type: {py_fn_type}"
 
+    def check_ref(self, index, arg, arg_strs):
+        if arg["tag"] == "ref":
+            if self.variableMap[arg["name"]].lower() == "real":
+                arg_strs[index] = arg_strs[index] + "._val"
+                return True
+            else:
+                return False
+        elif arg["tag"] == "op":
+            for item in arg["left"]:
+                st = self.check_ref(index, item, arg_strs)
+            if not st:
+                for item in arg["right"]:
+                    st = self.check_ref(index, item, arg_strs)
+            return st
+        elif arg["tag"] == "call":
+            for item in arg["args"]:
+                st = self.check_ref(index, item, arg_strs)
```

Comment:
add an "else" at the end of this if-elif sequence to catch anything not covered:
```
if arg["tag"] == "ref":
    ...
elif ...
   ...
elif ...
    ...
else:
    assert False
```